evalml/objectives/standard_metrics.py

```diff
@@ -901,6 +902,40 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class SilhouetteCoefficient(UnsupervisedLearningObjective):
+    """Silhouette Coefficient for clustering. Does not require ground truth values."""
+
+    name = "Silhouette Coefficient"
+    greater_is_better = True
+    score_needs_proba = False
+    perfect_score = 1.0
+    is_bounded_like_percentage = False
+    expected_range = [-1, 1]
+    requires_y_true = False
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        """Objective function for silhouette coefficient for clustering."""
+        try:
+            return metrics.silhouette_score(X, y_predicted, metric="euclidean")
+        except ValueError:
```

Comment:
Good point!

Reply:
The silhouette coefficient throws a `ValueError` in the case where there is only one cluster detected. Since other metrics (such as the rand score) simply return a score of 0 in this case, that's what this catch is trying to emulate.

It occurs to me now, I could implement this in a slightly safer and cleaner way:
```
if len(y_predicted) == 0:
    return 0
return metrics.silhouette_score(...)
```
This way any other ValueErrors will raise as necessary. Will update!